### ISSUE TITLE
Queries related to issues App features and Change requirements.

### ISSUE THREAD

**Issue by raghavendrakumbhar** [OfficeDev/microsoft-teams-apps-company-communicator#769]
Hi,

Request you to kindly suggest the options for below requirements.

1. While composing message by author in config tab, can we use the image url from our intranet Sharepoint site document library . Does it has to be a public folder?, If we want to use the app for intranet purpose then can't we use the image url from intranet repository or it needs to be public folder only.. Kindly suggest. 
2. Could you please explain a bit more about "Send in Chat to everyone" option
     Does it sends the message to everyone in the organization, it will be bit risky to give such option to authors. In the warning message it says the message will be sent to all who have client app. So, if we install the app with parameter **proactivelyInstallUserApp** set to true. in that case will it send message to everyone in the organization?. How do we prevent or restrict this option, should we install the app with the parameter **proactivelyInstallUserApp** set to false.

Please do share your feedback on these queries with a bit of explanation so that we are very clear on it. Thanks you very much in advance.

**Comment by gsv022** [comment]
Hi @raghavendrakumbhar ,
1)Image should be a public folder. If it is from intranet Sharepoint site all the users should have access to the image, then only the image will be displayed/delivered to all.
2)If proactive installation is set to true and user app is installed in that team/teams then the user app will installed to all users automatically and message will be sent to all of them  . If false, it can be restricted but the user app has to be installed explicitly for the user/users who wants to receive the message.

**Comment by raghavendrakumbhar** [comment]
Hi @gsv022,

Thanks for your response will check and let you know on the image requirement.

In case of send message to everyone option, say if we have 10 teams in ms-teams and the user app is installed on 5 of them. So, if we choose send message to everyone, will the message go to all members of 10 teams or all members of 5 teams where the user app is installed. We just want to know if that option sends message to everyone in the organization who have access to teams, as its a risk to send message to everyone in the organization?

**Comment by v-royavinash** [comment]
Hi @raghavendrakumbhar ,

Send to everyone is primarily focused to send messages across the tenant users. If proactive installation is enabled and user app is uploaded to tenant level then the app will be installed to respective users during message sending.

However to avoid this scenario, there are two alternatives,

- Turn off  proactive installation and manually install the user app using permission policies. In this messages will be send only to users for whom App is installed.

- Even if proactive installation is enabled, but if user app is not uploaded at tenant level, instead added to respective teams then the feature of proactive installation will not work and messages will be sent only for the users for whom the user app is manually installed.

**Comment by raghavendrakumbhar** [comment]
Hi @v-royavinash ,

Thank you very much for the alternative options, these would definitely help us.

While installing the app, we had kept proactive installation of aap setting as true, can we change that from the configuration of app service by setting it to false.

or, we shall keep the proactive setting as true and not upload the user app at tenant level, instead can we do side loading of the user app to only the teams where it is required. So, we if we side load the app to 5 teams. then users of only those 5 teams will receive message and not the rest of teams, right?

**Comment by v-royavinash** [comment]
Hi @raghavendrakumbhar ,
The steps you mentioned are valid and the suggested approach.
However to disable the proactive installation, the app setting of both app service and *-prep-function has to be updated.

**Comment by gsv022** [comment]
Hi @raghavendrakumbhar ,
Thank you for submitting this issue, as we work to ensure consistent responses with items logged in our issues list, we are working to address these items that did not get responses. Please let us know if you are still facing this issue and require assistance and we will be happy to discuss. If no further action is needed at moment and no response is received, we will close out the issue in 3 business days. You may also always open a new issue or respond here if something does come back up.

**Comment by raghav-nie** [comment]
Hi,

We tried below scenario today, its not behaving the way you explained above. We wanted to test "Send Message to all " scenario, which should send message to only those teams where user app is installed.

1. Installed the author app on a Team through side loading option
2. Installed the user app to 3 teams through side loading option
3. During Company Communicator app deployment, we have set Proactive Installation as true.
4. When we tried to send message using the option "Send Message to All", it took almost 9 minutes with status showing as installation in progress, so we cancelled the send message which was in progress. When we viewed the status it was showing it got failed for 1000 users and cancelled to around 1500 users. Actually we were testing it only for 3 teams which had hardly 25 users altogether. So, this means if the app is installed with Proactive installation as true the send message to all option sends it to everyone in organization, irrespective of it being installed at tenant level or not.
Kindly suggest what approach to use if we dont want to keep send message to all option to avoid sending it to everyone in organization.
